xla/service/gpu/autotuning/gemm_fusion_autotuner.h

```diff
@@ -198,6 +197,26 @@ class GemmFusionAutotunerImpl {
   std::vector<TritonGemmConfig> triton_configs_;
 };
 
+struct TileSizeLimit {
```

Comment:
All of the declarations below are currently directly in the `xla::gpu` namespace, but shouldn't be. The reason is that their name are too generic for top-level symbols.

Would moving these as members of `GemmFusionAutotuner` and/or `GemmFusionAutotunerImpl` also work?

Reply:
Also look at my comment on the new `_cuda` file. Perhaps we can simply revert the change here as well, as suggested in that comment.